cli/src/main/java/com/devonfw/tools/ide/merge/xmlmerger/XmlMerger.java

```diff
@@ -64,25 +62,51 @@ public void merge(Path setup, Path update, EnvironmentVariables resolver, Path w
       document = load(workspace);
     } else if (Files.exists(setup)) {
       document = load(setup);
+      target = setup;
     }
     if (updateFileExists) {
+      template = update;
       if (document == null) {
         document = load(update);
       } else {
         Document updateDocument = load(update);
-        merge(updateDocument, document);
+        merge(updateDocument, document, template, target);
       }
-      template = update;
     }
     resolve(document, resolver, false, template);
     save(document, workspace);
   }
 
-  public void merge(Document sourceDocument, Document targetDocument) {
+  public void merge(Document sourceDocument, Document targetDocument, Path source, Path target) {
+
+    this.context.debug("Merging {} with {} ...", source.getFileName().toString(), target.getFileName().toString());
+    MergeElement sourceRoot = new MergeElement(sourceDocument.getDocumentElement(), source);
+    MergeElement targetRoot = new MergeElement(targetDocument.getDocumentElement(), target);
+
+    if (matchRootElements(sourceRoot, targetRoot)) {
+      MergeStrategy strategy = MergeStrategy.of(sourceRoot.getMergingStrategy());
+      strategy.merge(sourceRoot, targetRoot, new ElementMatcher(this.context));
+    }
+  }
+
+  private boolean matchRootElements(MergeElement sourceRoot, MergeElement targetRoot) {
```

Comment:
Please add Javadoc.

Reply:
done, also changed the name of the method to something clearer.